File: prediction/extract-candidates.js
const fs = require('fs');
const path = require('path');
const htmlPath = path.join(process.env.USERPROFILE || '', 'Downloads', 'FBR Slides.html');
const html = fs.readFileSync(htmlPath, 'utf8');
const base = 'https://wwwmpa.mpa-garching.mpg.de/fbrslides/';
const re = /<img src="\.\/FBR Slides_files\/([^"]+)"[^>]*>.*?<h3>([^<]*)<\/h3>/g;
const out = [];
let m;
while ((m = re.exec(html)) !== null) {
  const filename = m[1];
  const name = m[2].trim();
  const imageUrl = base + 'FBR%20Slides_files/' + encodeURIComponent(filename);
  out.push({ id: String(out.length + 1), name, imageUrl });
}
fs.writeFileSync(path.join(__dirname, 'candidates.json'), JSON.stringify(out, null, 2));
console.log('Extracted', out.length, 'candidates to candidates.json');

File: prediction/extract_candidates.py
import re
import json
from urllib.parse import quote

path = r"C:\Users\Thibault Lechien\Downloads\FBR Slides.html"
html = open(path, encoding="utf-8").read()
base = "https://wwwmpa.mpa-garching.mpg.de/fbrslides/"
re_pat = re.compile(r'<img src="\./FBR Slides_files/([^"]+)"[^>]*>.*?<h3>([^<]*)</h3>')
out = []
for m in re_pat.finditer(html):
    filename, name = m.group(1), m.group(2).strip()
    image_url = base + "FBR%20Slides_files/" + quote(filename, safe="")
    out.append({"id": str(len(out) + 1), "name": name, "imageUrl": image_url})
json_path = r"c:\Users\Thibault Lechien\Documents\GitHub\Personal Website\thibaultLe.github.io\prediction\candidates.json"
open(json_path, "w", encoding="utf-8").write(json.dumps(out, indent=2, ensure_ascii=False))
print(len(out), "candidates written")
